src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    public AuthenticationServiceImpl() {
+
+    }
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDB = userService.findByEmail(email);
+        if (userFromDB.isPresent()) {
+            User user = userFromDB.get();
+            String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+            if (hashedPassword.equals(user.getPassword())) {
+                return user;
+            }
+        }
+        throw new AuthenticationException("Check your password and email. "
+                + "email: " + email
+                + ", password: " + password);
```

Comment:
When logging or throwing exceptions for authentication issues, it's best to keep the messages generic. The user should be informed that their login attempt failed, but you should not provide details that could aid an attacker, such as whether the email or the password was incorrect, and certainly not the actual password.